### DISCUSSION TITLE
Password

### DISCUSSION THREAD

**Discussion by Cyrille56** [pcmihnea/HA_WMR500#2] [Q&A] [answered]
Hi !

Thanks a lot for this incredibale work.

I can now connect my wmr500 to my wifi.

I have redirected app.idtlive.com and mqtt.idtlive.com to a pc (Ubuntu then Debian) from my wifi box (local dns).

The problem ils to collect the password with the command : sudo tshark -i eth0 -f "tcp port 1883" -Y 'mqtt.passwd' -V 
I have no return….

Any idea ?

Thank’s again 

Cyrille

**Comment by pcmihnea** [comment, marked as answer]
Hi, can you confirm with any sniffing tool, like Wireshark, that you're getting MQTT and/or HTTPS traffic from WMR500 to the PC?
Did you do the DNS configuration before connecting the WMR500 to WiFi? Otherwise you need to cold-boot it (remove batteries and power supply, then wait at least 10 seconds) to update the WMR500 DHCP info. 
Note that the tshark command targets interface `eth0` - modifiy based on your PC setup.

**Reply by Cyrille56** [reply]
Hi,
 With the command sudo tshark -i eth0 -f "tcp port 1883" I can see some traffic (more with sudo tshark -i eth0) and I have change eth0 with my ethernet interface name.
I have cold-booted too.
With ubuntu I installed homeassistant hoping some mqtt traffic from wmr500…. whithout success.
I will try again later with a fresh debian install.
Thanks again (Your « tuto » will help a lot or WMR500 owner)

**Reply by pcmihnea** [reply]
The tshark filters out only MQTT connect packets destined to a unsecured MQTT broker. Check with any MQTT client if your broker accepts unauthorized connections (should be by default for a fresh install of Mosquitto). Use the desktop version of Wireshark for easier visualization, and just filter by IP.addr==191.168.x.x of your WMR500.

**Reply by Cyrille56** [reply]
Fresh install mousqueton + wireshark + ip filter = a lot of packets….
With mqtt.passwd filter… none (over 14742 packets)

**Reply by pcmihnea** [reply]
EDIT: A valid MQTT broker configuration is necessary - setup authentication before.
EDIT2: I've updated the readme.md - maybe it'll help.

Just tried it on my setup - cold-booted the device, and after less than a minute the packet arrived. What type of MQTT packets are you seeing in Wirehark? Try with filter `-Y 'mqtt'` to see everything.
One possibility may be that your Mosquitto install doesn't allow unauthorised connections - https://mosquitto.org/documentation/authentication-methods/.
After setting it up, try connecting with for eg. http://mqtt-explorer.com/ from another PC, and publish anything, just to be sure your broker works.

**Reply by Cyrille56** [reply]
Got it !
I wrote a user with password in a file.
Also write listener 1883 and allow_anonymous true in the config file.
Still having problem to send command with mqtt explorer (not a mqtt user) : no change on the setting (no unit or language change)

**Reply by pcmihnea** [reply]
You need to publish a message via topic `enno/out/json/_GUUID_` (for example `enno/out/json/bb2ca03c-599b-407e-9239-ae135d07b315` as random GUUID), with the payload as JSON - `{"command": "setSettings", "ca1": "1", "id": "DEBUG"}` sets to Celsius. The response from WMR500 should be in the `enno/in/json`.
By the way, to check if the password's the same on all devices, does yours start with `80` and ends with `fc` like mine?

**Reply by Cyrille56** [reply]
Begin with c9 and ends with 34…. Sorry it should be too easy !!!
For the command, I did like you but enno/in/json doesn’t respond…

**Reply by Cyrille56** [reply]
Now enno/in/json return me the command but no change of units…

**Reply by pcmihnea** [reply]
The issue is that the device doesn't seem to publish anything without a request (i.e. `enno/out`) - check if it's actually connected via the `$SYS/broker/clients/connected` topic value, or use command `netstat -ntp | grep ESTABLISHED.*mosquitto`.
Try also the `{"command": "getChannel1Status", "id": "_GUUID_"}` payload.

**Reply by pcmihnea** [reply]
Note that 0=°F, 1=°C.

**Reply by Cyrille56** [reply]
I try to change language and wind speed

**Reply by pcmihnea** [reply]
What's the response payload?

**Reply by Cyrille56** [reply]
My WMR500 appears with netstat command

**Reply by Cyrille56** [reply]
Enno/in : {«type » : « m », « ts »:2022-02-24T17:33:33.350z……………}

**Reply by pcmihnea** [reply]
Mine returns something like {"type": "m","correlationId":"DEBUG"}","ts": "2022-02-24T18:48:00.000Z","deviceId":_GUUID_,"data":{"5":{"result":"true","desc":"success"}}} for setSettings command. What's the result/desc values?

**Reply by Cyrille56** [reply]
I’ve rebooted all…. And that’s work !!!!
You’re a Genius ! 

I would like to make donation for this project.Is there a way ?

Thank’s a lot

Cyrille

**Reply by pcmihnea** [reply]
Glad it was useful for you! There's no need for donations, the only help you could give is any info or details not yet covered, or at least confirm the validity of what's already documented :)

**Reply by Cyrille56** [reply]
Ok
Last message to confirm the message from enno/in : the same like yours 
 {"type": "m","correlationId":"DEBUG"}","ts": "2022-02-24T18:48:00.000Z","deviceId":GUUID,"data":{"5":{"result":"true","desc":"success"}}}
